### DISCUSSION TITLE
[FEAT] Protein AccNum mapping

### DISCUSSION THREAD

**Discussion by jananiravi** [JRaviLab/MolEvolvR#24] [Ideas]
- [ ] from UniProt to NCBI | `up2ncbi`
- [ ] NCBI to UniProt | `ncbi2up`
- [ ] Protein to IPG | `acc2ipg`
- [ ] other relevant ones?

Related
- [x] `acc2info`
- [ ] `tax2info`
- [ ] `acc2tax`
- [ ] `gca2info` | gpa --> assembly taxID --> info

[edirect and eutils e-book](https://www.ncbi.nlm.nih.gov/books/NBK179288/)

**Comment by Rhaenyra-Liang** [comment]
Hello @jananiravi , I would love to take on this task , Could you please assign this to me?

**Reply by jananiravi** [reply]
Great! Let us know if you have any Qs. Where would you like to start?

**Reply by Rhaenyra-Liang** [reply]
Thank you, @jananiravi! I have a few questions:

1. I understand this issue is about writing a tool, starting with a function called up2ncbi, which will convert UniProt to NCBI accession numbers, similar to the acc2fa file in the repo’s man folder. Is that correct?
2. After completing the first few items on the list, should I move on to other relevant ones and continue developing the tool?
3. I’m not sure what is meant by the “related list” because I cannot find a reference to it in the repo. 
4. Are there any clues that need to be retrieved through edirect and eutils (e-book link)?

I appreciate your help!

**Reply by jananiravi** [reply]
@Rhaenyra-Liang, the idea is to write `roxygen2`/`docstring` annotated functions with documentation. Check out our current functions that have some documentation (if you're unfamiliar with this). All the new functions will also be part of this MolEvolvR R package (and the companion web-app 2.0 down the line)! Does that answer your question? 
The current R package has been set up with `usethis`, `biocthis` (the bioconductor variation). Let me, @the-mayer know if you have any other questions.
In terms of where to start, the `eutils`, `entrez`, and the R implementation, `rentrez` would be a good place to start.

**Reply by Klangina** [reply]
Hello @jananiravi ,

Can I take this up?
Planning to start with UniProt <-> NCBI and Protien to IPG. Can move on later to the other related functions.

@Rhaenyra-Liang : Please confirm what you're working on currently.

**Reply by jananiravi** [reply]
Yes, check in with @Rhaenyra-Liang regarding where she is. Maybe you could work in parallel on a different function, @Klangina. Ask me/ @epbrenner if the function's premise (or set-up) is unclear.

**Reply by jananiravi** [reply]
Related to #122
